### ISSUE TITLE
Include prefixes in all PV's throughout codebase

### ISSUE THREAD

**Issue by evalott100** [PandABlocks/PandABlocks-ioc#133]
In the new PVI, we no longer pass a prefix here:

https://github.com/PandABlocks/PandABlocks-ioc/blob/6eadee5057eeae38fef75e2ba693d6d8ce17fe53/src/pandablocks_ioc/_pvi.py#L378

Instead we need to pass the whole PV wherever components are created:

https://github.com/PandABlocks/PandABlocks-ioc/blob/6eadee5057eeae38fef75e2ba693d6d8ce17fe53/src/pandablocks_ioc/_pvi.py#L123-L129

We should do the same with softioc. This will involve removing the assignement of a global prefix here:

https://github.com/PandABlocks/PandABlocks-ioc/blob/6eadee5057eeae38fef75e2ba693d6d8ce17fe53/src/pandablocks_ioc/ioc.py#L507-L508

and passing the prefix arg throughout the ioc builder so it can be prepended to the PV strings.

**Comment by evalott100** [comment]
> `GetRecordNames`

**Comment by evalott100** [comment]
Whoops, accidental close.

> If it's helpful, instead of refactoring to not use `SetDeviceName`, you can retrieve the PV prefix using the `GetRecordNames()` function:

Ah I was looking for a way to get the record name from whatever softioc record. I still think I'd personally opt for passing the whole PV. I think it's cleaner than getting the prefix of the first record PVI deals with in a loop and appending it to all the widgets.

One thing that I did just think of - we could make the `EpicsName` class be a dataclass with `seperator: str`, `prefix: str`, `block: str`, `field: str | None` attributes and then an `__str__` which converts them when we pass to things.

**Comment by AlexanderWells-diamond** [comment]
Yes, there currently exists no official way to extract a PV name from the RecordWrapper interface. Technically it is accessible via `<record>._RecordWrapper__device._name`, but that's two levels of private interfaces to go through. 

If it's a feature that would be useful, raise an enhancement issue against PythonSoftIOC?

But I do see an argument for changing the `EpicsName` type. It would be a fairly large change given how widely that type is passed around, but it sounds good.

**Comment by evalott100** [comment]
> If it's a feature that would be useful, raise an enhancement issue against PythonSoftIOC?

I'm unsure if I think it would be useful... On the one hand it could be nice for certain problems, on the other the ioc's datastructures should be designed to aggregate that information anyway. We shouldn't have to give all the initialized records to our PVI wrapper.

> But I do see an argument for changing the EpicsName type. It would be a fairly large change given how widely that type is passed around, but it sounds good.

Yep that'd definitely be a seperate issue.

**Comment by coretl** [comment]
Bear in mind that this is all a temporary fix until we move to FastCS, so fixing it in the easiest way possible will be fine

**Comment by evalott100** [comment]
> Bear in mind that this is all a temporary fix until we move to FastCS, so fixing it in the easiest way possible will be fine

Ah okay, we'll replace this issue with storing the prefix in the `Pvi` wrapper class and appending it to all the signals.
